reco_utils/recommender/sar/sar_singlenode.py

```diff
@@ -323,11 +329,96 @@ def recommend_k_items(self, test, top_k=10, sort_top_k=False, remove_seen=False)
         # drop invalid items
         return df.replace(-np.inf, np.nan).dropna()
 
+    def recommend_popular_items(self, top_k=10, sort_top_k=False):
+        test_users = pd.Series([0])
+        test_scores = np.array([self.item_frequencies])
+
+        # ensure we're working with a dense matrix
+        if isinstance(test_scores, sparse.spmatrix):
+            test_scores = test_scores.todense()
+
+        return self._get_top_k_items(
+            users=test_users, scores=test_scores, top_k=top_k, sort_top_k=sort_top_k
+        )
+
+    def recommend_similar_items(self, items, top_k=10, sort_top_k=False):
+        """Recommend top K items similar to the provided seed items
+
+        Args:
+            items (pd.DataFrame): dataframe with item, user (optional), and rating (optional) columns
+            top_k (int): number of top items to recommend
+            sort_top_k (bool): flag to sort top k results
+
+        Returns:
+            pd.DataFrame: sorted top k recommendation items
+        """
+
+        # convert item ids to indices
+        item_ids = items[self.col_item].map(self.item2index)
+
+        # if no ratings were provided assume they are all 1
+        if self.col_rating in items.columns:
+            ratings = items[self.col_rating]
+        else:
+            ratings = pd.Series(np.ones_like(item_ids))
+
+        # create local map of user ids
+        if self.col_user in items.columns:
+            test_users = items[self.col_user]
+            user2index = {x[1]: x[0] for x in enumerate(items[self.col_user].unique())}
+            user_ids = test_users.map(user2index)
+        else:
+            # if no user column exists assume all entries are for a single user
+            test_users = pd.Series(np.zeros_like(item_ids))
+            user_ids = test_users
+        n_users = user_ids.drop_duplicates().shape[0]
+
+        # generate pseudo user affinity using seed items
```

Comment:
I don't understand what the role of users is in this function. Why should users even enter the picture? Don't you just want to get the k largest entries from the item similarity matrix?

Reply:
this is so you can input multiple cold users, or multiple item seeds

Reply:
And what does this mean more precisely? How do you handle cold users, or item seeds? Without reading the code one cannot understand what these functions do from the docstring. 
What scenarios are you covering with this function? We have an item-item similarity matrix already computed with `fit()`. One scenario is to recommend similar items to a given set of items, which can be simply read from this matrix. But clearly you want to cover other scenarios too.

Reply:
i'll try to clarify the options in the docstrings